snuba/query/query_processor.py

```diff
@@ -21,6 +24,9 @@ class is supposed to provide one method that takes a query
     def process_query(self,
         query: Query,
         context_data: TQueryProcessContext,
+        request_settings: Mapping[str, bool],
```

Comment:
I am a bit skeptic about the idea of passing an untyped mapping of settings, letting the processor either update the settings or update a second untyped query_hints mapping for a few reasons:
1) Query processor should be used outside this initial phase of extension processing (like for optimizing joins), there stats may not make sense
2) I don't see a reason for request_settings and query_hints to be Mappings. We did that for extension because the extension processor it the code that needs to parse the json loaded data specific to that extension. It is not the case for settings, you are going to have one settings schema only.

Why don't you create a class RequestSettings (immutable and properly typed) maybe you can create a dataclass for this. Then you can create a second class (query hints) that is created starting from the settings and that is updated by extensions? Then you only pass query hints to the clickhouse query and throw request settings away before getting to clickhouse.

Reply:
Making a separate `RequestSettings` and `QueryHints` makes sense.

I feel we do still need to pass in the value of `turbo` because of [this](https://github.com/getsentry/snuba/pull/462/files#diff-bda53fd0523724650b51756c9136302fR92) conditional (don't feel like we can evaluate that conditional only based on `final`). I suppose `QueryHints` can also have `turbo` because it is a hint for the formation of the query. Does that line up with how you were thinking about it too?

Reply:
Maybe we should just get rid of the requestSettings class altogether, parse settings and create the Hints directly.
Is there a strong reason to keep both around ?

Reply:
Thinking more about this, I actually feel we could get rid of `QueryHints` and push `final` into `Query` because:
1. We need `consistent` and `debug` (currently in `RequestSettings`), which aren't hints for the creation of the query. They're...something else. They're settings on the request which affect the where we run the query and what information is returned in the response.
1. The big reason for `QueryHints` is that we need a way to modify and pass down `final` to the clickhouse query. `final` is a hint but could also be put into `Query` cause it does affect the query being produced.